office_space_allocation/amity.py

```diff
@@ -0,0 +1,150 @@
+                rm = random.choice(offices)
+
+                rm.add_person(person)
+                self.allocated_rooms.append(rm)
+                return rm
+        elif isinstance(person, Fellow):
+            # raise exception if there are no rooms
+            if len(self.all_rooms) < 1:
+                raise IndexError("There are no available rooms. Please create some rooms before making allocations.")
+            else:
+                office_rm = random.choice(self.all_rooms)
+
+                office_rm.add_person(person)
+                self.allocated_rooms.append(office_rm)
+                return office_rm
+        else:
+            raise TypeError("Person argument must be of type Staff or Fellow")
+
+    def find_person(self, name):
+        """
+        Finds a Person using name.
+        Name can be the full name, first name, last name or part of either
+        :param name:
+        :return: a tuple with Person(s) objects found in the list if persons
+        """
+        name = name.strip()
+        res = [p for p in self.all_persons if name.title() in p.get_full_name()]
+        if not res:
+            raise Exception("Person was not found in the system")
+        else:
+            return tuple(res)
+
+    def reallocate_person(self, person_name, room_name):
+        """
+        Reallocates `Person` to another `Room` if possible
+
+        :param: person_name: Name of the `Person`
+        :param: room_name: Name of the `Room`
+        :return: a tuple with (`Person`, `Room`) if reallocation is successful
+        """
+        person_name = person_name.strip()
+        room_name = room_name.strip()
+
+        try:
+            # only one person should be found
+            people_found = self.find_person(person_name)
+            if len(people_found) > 1:
+                raise MultiplePeopleFoundException
+            else:
+                person = people_found[0]
+        except Exception:
+            raise
+
+        try:
+            room = self.find_room(room_name)
+        except ValueError:
+            raise
+
+        if room.has_person(person):
+            raise Exception("Cannot reallocate person to same room.\n {} is already in {}".format(person_name,
+                                                                                                  room_name))
+
+        # remove person in current room
+        found = False
+        for r in self.all_rooms:
+            if r.has_person(person):
+                found = True
+                r.remove_person(person)
+                break
+        if not found:
+            raise Exception("{} has not been added to any room.\n Please add them to a room before making any "
+                            "reallocation".format(person_name))
+
+        # add person to room
+        try:
+            room.add_person(person)
+        except InvalidRoomOccupantError:
+            raise
+        except RoomFullError:
+            raise
+        return person, room
```

Comment:
Nitpick: Why not return a more understandable statement, like `person + " has been added to" + room`. Because you're using it in the command line